scripts/main/upload.js

```diff
@@ -918,3 +918,220 @@ upload.check = function () {
 		$resync.prop("checked", false).prop("disabled", true);
 	}
 };
+
+/**
+ * @param {(FileList|File[])} files
+ */
+upload.uploadTrack = function (files) {
```

Comment:
I believe this method can heavily be simplified as you only can add a single track to an album.

Reply:
I think I did this as far as possible, but most other methods (especially `upload.show`) are built for lists and require them. Most other usages of this list are as simple as possible.

Reply:
I'll take a crack at simplifying it further...

Reply:
@kamil4 Did you change anything?

Reply:
"Changed" doesn't accurately capture the extent of what I did to this code. More like "decimated", really :wink:.

Reply:
And still my IDE complains about 40 lines of duplicated code. But I did not want to take care of that. When the “Import from URL” will also use streamed responses one day, then we should clean that up. (dreaming)